source/game/kart/KartDynamics.cc

```diff
@@ -178,8 +189,44 @@ void KartDynamics::setAngVel0(const EGG::Vector3f &v) {
     m_angVel0 = v;
 }
 
+void KartDynamics::setAngVel2(const EGG::Vector3f &v) {
+    m_angVel2 = v;
+}
+
 void KartDynamics::setAngVel0YFactor(f32 val) {
     m_angVel0YFactor = val;
 }
 
+void KartDynamics::setTop_(const EGG::Vector3f &v) {
+    m_top_ = v;
+}
+
+KartDynamicsBike::KartDynamicsBike() = default;
+
+KartDynamicsBike::~KartDynamicsBike() = default;
+
+void KartDynamicsBike::forceUpright() {
+    m_angVel0.z = 0.0f;
+}
+
+void KartDynamicsBike::stabilize() {
+    EGG::Vector3f forward = m_mainRot.rotateVector(EGG::Vector3f::ez);
+    EGG::Vector3f cross = m_top_.cross(forward);
+    EGG::Vector3f forward_ = cross.cross(m_top_);
+    forward_.normalise();
+    EGG::Vector3f local_40 = m_top_.cross(forward_);
+    EGG::Vector3f local_4c = forward_.cross(local_40);
+    local_4c.normalise();
```

Comment:
```suggestion
    EGG::Vector3f forward = m_top_.cross(m_mainRot.rotateVector(EGG::Vector3f::ez)).cross(m_top_);
    forward.normalise();
    EGG::Vector3f local_4c = forward.cross(m_top_.cross(forward));
    local_4c.normalise();
```

I'm fine with splitting these up as needed, but it does eliminate the explicit definition of a few stack variables.